Decimation/Simplify.h

```diff
@@ -87,942 +83,977 @@ struct vec3f
-	double v = (d11 * d20 - d01 * d21) / denom;
-	double w = (d00 * d21 - d01 * d20) / denom;
-	double u = 1.0 - v - w;
-	return vec3f(u,v,w);
+    vec3f v0 = b-a;
+    vec3f v1 = c-a;
+    vec3f v2 = p-a;
+    double d00 = v0.dot(v0);
+    double d01 = v0.dot(v1);
+    double d11 = v1.dot(v1);
+    double d20 = v2.dot(v0);
+    double d21 = v2.dot(v1);
+    double denom = d00*d11-d01*d01;
+    double v = (d11 * d20 - d01 * d21) / denom;
+    double w = (d00 * d21 - d01 * d20) / denom;
+    double u = 1.0 - v - w;
+    return vec3f(u,v,w);
 }
 
 vec3f interpolate(const vec3f &p, const vec3f &a, const vec3f &b, const vec3f &c, const vec3f attrs[3])
 {
-	vec3f bary = barycentric(p,a,b,c);
-	vec3f out = vec3f(0,0,0);
-	out = out + attrs[0] * bary.x;
-	out = out + attrs[1] * bary.y;
-	out = out + attrs[2] * bary.z;
-	return out;
+    vec3f bary = barycentric(p,a,b,c);
+    vec3f out = vec3f(0,0,0);
+    out = out + attrs[0] * bary.x;
+    out = out + attrs[1] * bary.y;
+    out = out + attrs[2] * bary.z;
+    return out;
 }
 
 double min(double v1, double v2) {
-	return fmin(v1,v2);
+    return fmin(v1,v2);
 }
 
 
 class SymetricMatrix {
 
-	public:
-
-	// Constructor
-
-	SymetricMatrix(double c=0) { loopi(0,10) m[i] = c;  }
-
-	SymetricMatrix(	double m11, double m12, double m13, double m14,
-			            double m22, double m23, double m24,
-			                        double m33, double m34,
-			                                    double m44) {
-			 m[0] = m11;  m[1] = m12;  m[2] = m13;  m[3] = m14;
-			              m[4] = m22;  m[5] = m23;  m[6] = m24;
-			                           m[7] = m33;  m[8] = m34;
-			                                        m[9] = m44;
-	}
-
-	// Make plane
-
-	SymetricMatrix(double a,double b,double c,double d)
-	{
-		m[0] = a*a;  m[1] = a*b;  m[2] = a*c;  m[3] = a*d;
-		             m[4] = b*b;  m[5] = b*c;  m[6] = b*d;
-		                          m[7 ] =c*c; m[8 ] = c*d;
-		                                       m[9 ] = d*d;
-	}
-
-	double operator[](int c) const { return m[c]; }
-
-	// Determinant
-
-	double det(	int a11, int a12, int a13,
-				int a21, int a22, int a23,
-				int a31, int a32, int a33)
-	{
-		double det =  m[a11]*m[a22]*m[a33] + m[a13]*m[a21]*m[a32] + m[a12]*m[a23]*m[a31]
-					- m[a13]*m[a22]*m[a31] - m[a11]*m[a23]*m[a32]- m[a12]*m[a21]*m[a33];
-		return det;
-	}
-
-	const SymetricMatrix operator+(const SymetricMatrix& n) const
-	{
-		return SymetricMatrix( m[0]+n[0],   m[1]+n[1],   m[2]+n[2],   m[3]+n[3],
-						                    m[4]+n[4],   m[5]+n[5],   m[6]+n[6],
-						                                 m[ 7]+n[ 7], m[ 8]+n[8 ],
-						                                              m[ 9]+n[9 ]);
-	}
-
-	SymetricMatrix& operator+=(const SymetricMatrix& n)
-	{
-		 m[0]+=n[0];   m[1]+=n[1];   m[2]+=n[2];   m[3]+=n[3];
-		 m[4]+=n[4];   m[5]+=n[5];   m[6]+=n[6];   m[7]+=n[7];
-		 m[8]+=n[8];   m[9]+=n[9];
-		return *this;
-	}
-
-	double m[10];
+    public:
+
+    // Constructor
```

Comment:
Indentation should be 2 spaces instead of 4.

Reply:
> The styling in this file is strange in multiple places

Indeed, I also added 4 spaces to be consistent ... I will run astyle in a follow up PR.